File: fix_nav_logo.js
const fs = require('fs');
const path = require('path');

const root = 'c:\\Users\\lynns\\Downloads\\detecthiddenfees.com';

function findHtmlFiles(dir) {
    const results = [];
    try {
        const items = fs.readdirSync(dir, { withFileTypes: true });
        for (const item of items) {
            if (item.name.startsWith('.') || item.name === 'node_modules') continue;
            const fullPath = path.join(dir, item.name);
            if (item.isDirectory()) {
                results.push(...findHtmlFiles(fullPath));
            } else if (item.name.endsWith('.html')) {
                results.push(fullPath);
            }
        }
    } catch (e) {}
    return results;
}

const htmlFiles = findHtmlFiles(root);
console.log('Total HTML files: ' + htmlFiles.length);

const newNav = '<nav><div class="container nav-wrap"><a href="/" style="text-decoration:none;color:inherit"><div class="logo">Detect<span>HiddenFees</span></div></a><div class="nav-links"><a href="/ai-analysis-hub.html">AI Tools</a><a href="/hidden-fees-guides.html">Guides</a><a href="/ai-contract-review.html">Contract Review</a><a href="/ai-bill-analyzer.html">Bill Analyzer</a><a href="/ai-financial-advisor.html">AI Financial Advisor</a></div><a href="https://hiddenfeeai.com" rel="noopener noreferrer" class="nav-btn">Analyze My Document →</a></div></nav>';

// The exact old header pattern from the investigation pages
const oldHeader = '<header style="position:sticky;top:0;z-index:999;background:rgba(2,6,23,.85);backdrop-filter:blur(24px);border-bottom:1px solid rgba(59,130,246,0.15);padding:14px 20px"><a href="/" style="text-decoration:none;color:inherit"><div class="logo"> DetectHiddenFees </div></a></header>';

let navFixed = 0;
let logoFixed = 0;
let alreadyGood = 0;

for (const f of htmlFiles) {
    try {
        let c = fs.readFileSync(f, 'utf8');
        const o = c;
        
        // Skip index.html
        if (f.endsWith('index.html') || f.endsWith('index.html')) continue;
        
        // Check if already has the full nav with AI Financial Advisor
        if (c.includes('nav-wrap') && c.includes('ai-financial-advisor')) {
            alreadyGood++;
            continue;
        }
        
        // Replace old bare header with full nav
        if (c.includes(oldHeader)) {
            c = c.replace(oldHeader, newNav);
            navFixed++;
            console.log('NAV: ' + path.relative(root, f));
        }
        
        // Fix bare logo text
        if (c.includes('>DetectHiddenFees</div>') && !c.includes('Detect<span>HiddenFees</span>')) {
            c = c.replace(/>DetectHiddenFees<\/div>/g, '>Detect<span>HiddenFees</span></div>');
            logoFixed++;
            if (!c.includes(oldHeader)) {
                console.log('LOGO: ' + path.relative(root, f));
            }
        }
        
        if (c !== o) {
            fs.writeFileSync(f, c, 'utf8');
        }
    } catch (e) {
        if (!e.message.includes('EISDIR')) {
            console.log('ERR: ' + path.relative(root, f) + ' - ' + e.message.substring(0, 60));
        }
    }
}

console.log('\nNavs replaced: ' + navFixed);
console.log('Logos fixed: ' + logoFixed);
console.log('Already had full nav: ' + alreadyGood);
console.log('Done!');

File: fix_nav_logo.py
import os, glob, re
from pathlib import Path

root = r'c:\Users\lynns\Downloads\detecthiddenfees.com'
os.chdir(root)

html_files = glob.glob('**/*.html', recursive=True)
print(f'Total HTML files found: {len(html_files)}')

files_updated = 0
files_with_old_nav = 0

for f in html_files:
    if 'node_modules' in f or '.git' in f:
        continue
    try:
        # Read file as text since we need to inspect content
        with open(f, 'r', encoding='utf-8', errors='ignore') as fh:
            content = fh.read()
        
        original = content
        
        # 1. CHECK: Does this file have the old nav (has nav-links but NO AI Financial Advisor link)?
        has_nav = 'nav-links' in content and 'Bill Analyzer' in content
        has_financial = 'ai-financial-advisor' in content
        
        if has_nav and not has_financial:
            files_with_old_nav += 1
            # Add AI Financial Advisor link after Bill Analyzer in nav
            content = content.replace(
                '<a href="/ai-bill-analyzer.html">Bill Analyzer</a></div>',
                '<a href="/ai-bill-analyzer.html">Bill Analyzer</a><a href="/ai-financial-advisor.html">AI Financial Advisor</a></div>'
            )
        
        # 2. Standardize nav CTA button text
        content = content.replace(
            'class="nav-btn">Analyze My Documents With AI →</a>',
            'class="nav-btn">Analyze My Document →</a>'
        )
        
        # 3. Standardize sticky CTA button text
        content = content.replace(
            'class="sticky-btn">Analyze My Documents Now</a>',
            'class="sticky-btn">Analyze My Document</a>'
        )
        content = content.replace(
            'class="sticky-btn">Analyze Documents Now</a>',
            'class="sticky-btn">Analyze My Document</a>'
        )
        
        # 4. Standardize cta-anchor button text
        content = content.replace(
            'Analyze My Documents With AI — $15',
            'Analyze My Document — $15'
        )
        
        # 5. Skip index.html (already has the full nav correctly)
        if f == 'index.html':
            continue
        
        if content != original:
            with open(f, 'w', encoding='utf-8') as fh:
                fh.write(content)
            files_updated += 1
            reason = 'nav+cta' if has_nav and not has_financial else 'cta only'
            print(f'  UPDATED ({reason}): {f}')
            
    except Exception as e:
        print(f'  ERROR {f}: {e}')

print(f'\\nFiles with old nav: {files_with_old_nav}')
print(f'Files updated: {files_updated}')
print('Done!')
